File: java/extentreports/src/main/java/com/relevantcodes/extentreports/IExtentTestClass.java
package com.relevantcodes.extentreports;

import java.util.Date;

import com.relevantcodes.extentreports.model.ITest;

public interface IExtentTestClass {
	void log(LogStatus logStatus, String stepName, String details);

	void log(LogStatus logStatus, String details);

	void log(LogStatus logStatus, String stepName, Throwable t);

	void log(LogStatus logStatus, Throwable t);

	void setDescription(String description);
	
	String getDescription();
	
	void setStartedTime(Date startedTime);
	
	Date getStartedTime();
	
	void setEndedTime(Date endedTime);
	
	Date getEndedTime();
	
	String addScreenCapture(String imgPath);

	String addScreencast(String screencastPath);

	ExtentTest assignCategory(String... categories);

	ExtentTest assignAuthor(String... authors);

	ExtentTest appendChild(ExtentTest node);

	LogStatus getRunStatus();

	ITest getTest();
}

File: dotnet/RelevantCodes.ExtentReports/RelevantCodes.ExtentReports/IExtentTestClass.cs
using System;

namespace RelevantCodes.ExtentReports
{
    interface IExtentTestClass 
    {
	    void Log(LogStatus Status, string StepName, string Details);

	    void Log(LogStatus Status, string Details);

	    void Log(LogStatus Status, string StepName, Exception Ex);

	    void Log(LogStatus Status, Exception Ex);

	    string AddScreenCapture(string ImgPath);

	    string AddScreencast(string ScreencastPath);

	    ExtentTest AssignCategory(params string[] Categorie);

	    ExtentTest AssignAuthor(params string[] Author);

	    ExtentTest AppendChild(ExtentTest Node);

	    LogStatus GetCurrentStatus();

        string Description { get; set; }

        DateTime StartTime { get; set; }

        DateTime EndTime { get; set; }
    }
}
